### PR TITLE
Use controller runtime

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

File: controller.go [hunk 1]
```diff
@@ -35,205 +32,79 @@ const (
...
-		}
-	case *v1.ConfigMap:
-		for _, val := range c.ConfigMapWatchIgnore {
-			if _, exists := object.Annotations[val]; exists {
-				return true
-			}
-		}
-	default:
-		return false
+	if ingressClass != r.IngressClass {
```
Comment (nettoclaudio) [inline controller.go:62]: Now that we're using Ingress under API version networking.k8s.io/v1, we should give precedence to [`Ingress.Spec.IngressClassName`](https://pkg.go.dev/k8s.io/api/networking/v1#IngressSpec) first and then fall back to Ingress class annotation while validating the Ingress class.

File: controller.go [hunk 2]
```diff
@@ -35,205 +32,79 @@ const (
...
-	for _, ingressIface := range c.ingressesIndex.List() {
-		ingress := ingressIface.(*extensionsV1beta1.Ingress)
-
+	for _, ingress := range ingresses.Items {
 		ingressClass := ingress.Annotations[IngressClassAnnotation]
-		if ingressClass != c.IngressClass {
-			if _, exists := ingress.Annotations[ResultAnnotation]; exists {
-				orphaned[ingress.Namespace+"/"+ingress.Name] = ingress
-			}
+		if ingressClass != r.IngressClass {
```
Comment (nettoclaudio) [inline controller.go:94]: Same as above, we should validate the IngressClassName field before.

File: controller.go [hunk 3]
```diff
@@ -243,240 +114,279 @@ func (c *Controller) Process(ctx context.Context) {
...
-				glog.Errorf("Could unmarshal [%s] from ConfigMap [%s/%s]: %v", LabelsConfigKey, configMap.Namespace, configMap.Name, err)
+	var (
+		ownerReferences []metaV1.OwnerReference
+		tls             []networkingv1.IngressTLS
+		rules           []networkingv1.IngressRule
+	)
+
+	for _, ingress := range ingresses {
+		ownerReferences = append(ownerReferences, metaV1.OwnerReference{
+			APIVersion: "extensions/v1beta1",
```
Comment (nettoclaudio) [inline controller.go]:
```suggestion
			APIVersion: ingress.APIVersion,
```

File: controller.go [hunk 4]
```diff
@@ -243,240 +114,279 @@ func (c *Controller) Process(ctx context.Context) {
...
-			existingMergedIngress := existingMergedIngressIface.(*extensionsV1beta1.Ingress)
+	mergedIngress := &networkingv1.Ingress{
+		ObjectMeta: metaV1.ObjectMeta{
+			Namespace:       configMap.Namespace,
+			Name:            name,
+			Labels:          labels,
+			Annotations:     annotations,
+			OwnerReferences: ownerReferences,
+		},
+		Spec: networkingv1.IngressSpec{
```
Comment (nettoclaudio) [inline controller.go:290]: We should add the class name here.

**Review by nettoclaudio — CHANGES_REQUESTED** [review]
I left some tiny comments regarding the change to new Ingress' API version. The MR looks good to me overall.

File: controller.go (hunk 3 above)
Comment (wpjunior) [reply controller.go]: nice catch @nettoclaudio

File: controller.go (hunk 4 above)
Comment (wpjunior) [reply controller.go:290]: Done @nettoclaudio

File: controller.go (hunk 1 above)
Comment (wpjunior) [reply controller.go:62]: nice catch @nettoclaudio

**Pull request merged**